TypeCobol/Analysis/AnalyzerProviderLoader.cs

```diff
@@ -17,19 +19,28 @@ public static class AnalyzerProviderLoader
         /// </summary>
         /// <param name="compositeAnalyzerProvider">CompositeAnalyzerProvider instance into which the external analyzers should be added.</param>
         /// <param name="assemblyFilePaths">List of paths of .NET Assembly files.</param>
-        public static void AddCustomProviders(this CompositeAnalyzerProvider compositeAnalyzerProvider, IEnumerable<string> assemblyFilePaths)
+        /// <param name="logFilePath">Path of a non-null log file.</param>
+        public static void AddCustomProviders(this CompositeAnalyzerProvider compositeAnalyzerProvider, IEnumerable<string> assemblyFilePaths, [NotNull] string logFilePath)
         {
             if (compositeAnalyzerProvider == null || assemblyFilePaths == null) return;
 
             foreach (var assemblyFilePath in assemblyFilePaths)
             {
-                var provider = LoadProvider(assemblyFilePath);
-                compositeAnalyzerProvider.AddProvider(provider);
+                try
+                {
+                    var provider = LoadProvider(assemblyFilePath);
+                    compositeAnalyzerProvider.AddProvider(provider);
+                }
+                catch (Exception exception)
+                {
+                    File.AppendAllText(logFilePath, $"Failed to load analyzer provider on path {assemblyFilePath}\n{exception.Message}\n{exception}");
+                }
             }
         }
 
         /// <summary>
         /// Look for a suitable type into the assembly given and instantiate a new IAnalyzerProvider from it.
+        /// <remarks>Caller must catch potential exceptions.</remarks>
```

Comment:
Absolutely ! We can even emphasize this by renaming the method to `UnsafeLoadProvider` ?

Reply:
That's a good idea !